test/Listener/ConfigListenerTest.php

```diff
@@ -370,4 +370,99 @@ public function testConfigListenerFunctionsAsAggregateListener()
         $configListener->detach($events);
         $this->assertEquals(2, count($this->getEventsFromEventManager($events)));
     }
+
+    public function datasetCachedConfigs()
+    {
+        $datasets = [
+            'valid_file' => [
+                // Test for check if cache is properly used if is valid
+                ['data' => 'any'], // expects to fall back to loading all modules
+                '<?php return [\'data\' => \'any\'];',
+            ],
+            'inexistent_file' => [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                null, // file won't be created
+            ],
+            'file_with_data_before_php_tag' => [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                'something<?php return [\'data\' => \'any\'];',
+            ],
+            'malformed_open_tag' => [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                '<\?php return [\'data\' => \'any\'];',
+            ],
+        ];
+        if (version_compare(PHP_VERSION, '7.0', '>=')) {
```

Comment:
It's a little problematic as from my knowledge you cannot include config cache file with catching all errors like syntax etc.  Some time ago there was php_check_syntax() function in PHP 5 but it's deprected since 5.0.5 so i think that it's impossible to create fix for this problem which will work in 5.X.

Reply:
I've tried it - and when I include this case and run tests on PHP 5.6 all works as expected (test still passes)

Reply:
I have added it after failure from Travis, f.x.:
https://travis-ci.org/zendframework/zend-modulemanager/jobs/557784100

Reply:
Ok, you are right, checked again and getting the same error. Thx 👍